FIREFLY-1795: fix alert banner for both light and dark mode

kpuriIpac opened the pull request:
**Ticket**: [FIREFLY-1795](https://jira.ipac.caltech.edu/browse/FIREFLY-1795) 
- @robyww  I checked with @loitly  - IRSA doesn't have a reason for the current hardcoded background color for the alerts banner, so removing the old background color allows JoyUI to use the default warning color which works for both dark & light mode as expected. 
  - I experimented with `soft` & `solid` variants, I think `soft` looks better (and also similar to what we used to have before)

**Testing**: 
- **Euclid**: https://firefly-1795-alert-banner.irsakudev.ipac.caltech.edu/applications/euclid
- **Firefly**: https://fireflydev.ipac.caltech.edu/firefly-1795-alert-banner/firefly
- Just see the alert banners on both light and dark modes.  
  - I'll remove the hardcoded `msg` string before merging this PR.

kpuriIpac committed:
FIREFLY-1795: Fix alert banner for both light and dark mode

robyww commented:
The code looks good so far but I would also like you to tweek if so it better supports html.
```html
<style>
    .myWarning{
        background-color: var(--joy-palette-primary-softBg);
        text-align: center;
        font-size:var(--joy-fontSize-lg);
        margin-top: 5px
    }
    div div:has(.myWarning) {
        padding:0;
    }
    div:has(.myWarning) {
        width: 100%;
    }
</style>
<div class= 'myWarning'>
    alert: be very afraid
</div>
```
Rubin had to use something like the above. Notice the two `has` entries. I would prefer that they did not have to do something this advanced.  Also the `.myWarning` style should now be unnecessary but you should probably check. 

They care about using html because they want to add links in there alerts.

If you need to test for html there is a `TableUtil.js`, `isHtml()`

kpuriIpac commented:
@robyww I addressed your changes and updated the test builds (they now include a dummy link as well). Can you take a look and let me know what you think? 

It handles links when present in the HTML, and generally added a bit more generic styling, so hopefully Rubin won't need to change much else (but I added `ff-alert` and `ff-alert-content` classes if they would **_really_** like to still customize this further, if you think that's unnecessary I can remove those).

robyww commented on src/firefly/js/ui/DropDownContainer.jsx:
```diff
@@ -111,22 +112,51 @@ DropDownContainer.propTypes = {
+
+    let safeHtml;
+    if (looksHtml) {
+        safeHtml = cleanHtml(msg);
```
I have mixed feeling about using `cleanHtml`. It is safer but this is not user exposed html. It strips any style block. That give the deployment less control. There is not point then of having styles like `ff-alert-content` if we use `cleanHtml`

kpuriIpac replied:
Makes sense, what do you think the final change should be? Keep the call to `cleanHtml` and get rid of styles? (`ff-alert`)

loitly replied:
I was expecting @robyww  to respond since this was his request.
@kpuriIpac  I think you misunderstood.  He meant that you should not add `cleanHtml`, as it would strip out any embedded styling in the HTML. Without that styling, `ff-alert` wouldn’t serve its purpose. The right thing to do is remove `cleanHtml` and keep `ff-alert`.

robyww commented on src/firefly/js/ui/DropDownContainer.jsx:
```diff
@@ -111,22 +112,51 @@ DropDownContainer.propTypes = {
+    let safeHtml;
+    if (looksHtml) {
+        safeHtml = cleanHtml(msg);
+        safeHtml = enforceLinkTargets(safeHtml); //only touches <a> tags
```
I really like this idea.

robyww commented on src/firefly/js/ui/DropDownContainer.jsx:
```diff
@@ -111,22 +112,51 @@ DropDownContainer.propTypes = {
+            }}
+        >
+            {/* eslint-disable-next-line react/no-danger */}
+            <div className='ff-alert-content'  dangerouslySetInnerHTML={{ __html: safeHtml }} />
```
make this `div` to be `width:100%`. That would fix most need to have styles.

robyww commented on src/firefly/js/ui/DropDownContainer.jsx:
```diff
@@ -111,22 +112,51 @@ DropDownContainer.propTypes = {
+            .replace(/&/g, '&amp;')
+            .replace(/</g, '&lt;')
+            .replace(/>/g, '&gt;')
+            .replace(/\n/g, '<br/>');
```
good idea

robyww reviewed (approved):
For details see my comments below. I don't think we want to use safeHtml in this case. Make the cleanups and I think it is ready to merge.

The alert UI is much better now.

loitly commented:
> I noticed there’s a gap between the alerts and the rest of the page, which is especially noticeable in SPHEREx. We should address this.

**Never mind.  The gap is coming from SPHEREx. With the new lighter color and no border, it just blends into the white background more.**

Also, the additional enforcement (e.g., enforceLinkTargets) may not be necessary and could even be restrictive. In our use case, the admin is the one creating the message, so they should have control over the link target if needed.

kpuriIpac commented:
> Also, the additional enforcement (e.g., enforceLinkTargets) may not be necessary and could even be restrictive. In our use case, the admin is the one creating the message, so they should have control over the link target if needed.

Got it. I can get rid of `safeHtml = enforceLinkTargets(safeHtml);`, the idea was to ensure if there's a link, we set `target` to `_blank` (so that it opens a new tab always - in case the link that the admin puts in doesn't have this attribute set), but I can remove it too.

kpuriIpac commented:
@loitly @robyww I implemented the feedback. Let me know what you think. Any final changes?
